### PR TITLE
improve rendering depth image

### LINKED ISSUE #368: Return image depth from camera
Currently, the `image_depth` attribute of plotter returns a depth mapping from 0 to 1. We should fix this to return a true mapping of the distance from the camera to the meshes in the scene. A solution for this is posted on the VTK discourse: https://discourse.vtk.org/t/absolute-values-for-depth-image/326/5?u=banesullivan

https://github.com/pyvista/pyvista/blob/78e65f544f102ff9f488190a2938c68e4c89c55a/pyvista/plotting/plotting.py#L2578

Should be updated to:

```py
relative = self._run_image_filter(ifilter)
near, far = self.camera.GetClippingRange()
z_val = 2*near*far/((relative[:, :, 0] - 0.5) *2*(far - near)  - near - far)
z_val[z_val <= -far] = np.nan
return z_val
```

Ping @tomtomatron

### PR DESCRIPTION
**Pull request by tomtomatron**
Resolve #368 

In this pull request:

- pyvista.Plotter().image_depth raises an error now, it is replaced by get_image_depth()
- Added an example of using pyvista.Plotter().get_image_depth()  
- Updated the example readme

### CODE REVIEW

File: examples/02-plot/image_depth.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+"""
+Render a depth image
+~~~~~~~~~~~~~~~~~
```
Comment (banesullivan) [inline examples/02-plot/image_depth.py]: Make sure `~` matches the length of the title

File: examples/02-plot/image_depth.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from pyvista import examples
+
+# Load an interesting example of geometry
+mesh = examples.load_random_hills()
+
+# Establish geometry within a pv.Plotter()
+pv.close_all()
+p = pv.Plotter()
+p.add_mesh(mesh, color=True)
+p.show(auto_close=False)
```
Comment (banesullivan) [inline examples/02-plot/image_depth.py]: We need to be sure to close the plotter (`p.close()`) after we are done utilizing it

File: examples/02-plot/image_depth.py (hunk 1 above)
Comment (tomtomatron) [reply examples/02-plot/image_depth.py]: Done.

File: examples/02-plot/image_depth.py (hunk 2 above)
Comment (tomtomatron) [reply examples/02-plot/image_depth.py]: Done.  See 19c2b594767cc8cab3ab495822111b90b94473ab

**Comment by banesullivan** [comment]
I edited the test case and manually merged this

**Pull request closed without merging**